### PR TITLE
feat: add search results and dialog with input (resolves #207)

### LINKED ISSUE #207: Browse all page: When filters & search term is applied
Mockups now in VD Core Screens V02 CL folder for all viewports.

Preview:
Screen Shot 2020-01-28 at 3 14 02 PM

### PR DESCRIPTION
**Pull request by greatislander**
* [x] I've read the [guidelines for Contributing to the Platform Co-op Toolkit](https://github.com/platform-coop-toolkit/.github/blob/master/CONTRIBUTING.md)
* [x] This isn't a duplicate of an existing pull request

## Description

- [x] Adds support for optional text input to dialog component.
- [x] Implements page layout for search results.

## Steps to test

Review dialog component and search results layout.

## Additional information

Not applicable.

## Related issues

- Resolves #207.

### CODE REVIEW

**Comment by cherylhjli** [comment]
Spacing is good.
For modal with input, can we try this copy?

H1: Save Search
Body: To save your search, please give it a name so you can identify it later.

Input field label: Name of saved search:

CTA: 
Primary: Save
Secondary: Don’t save

**Comment by greatislander** [comment]
@cherylhjli Back to you for final review.

File: src/assets/scripts/Pinecone/Dialog/index.js [hunk 1]
```diff
@@ -61,6 +72,15 @@ class Dialog {
 		if ( this.config.question ) {
 			innerHtml += `<p id="q-${unique + 1}">${this.config.question}</p>`;
 		}
+		if ( this.config.input && this.config.inputLabel ) {
+			innerHtml += '<div class="input-group">';
+			innerHtml += `<label for="${this.config.input}">${this.config.inputLabel}</label>`;
+			innerHtml += `<input id="${this.config.input}" type="text" name="${this.config.input}" />`;
```
Comment (jhung) [inline src/assets/scripts/Pinecone/Dialog/index.js]: Couple questions (more design related):
* Should there be an explicit close button on the dialog? My expectation is there should be one.
* If the input field is mandatory, I think the initial focus should be on the input field (or second in the tab order if a close button is added as per comment above).
* If the input field is optional, the initial focus can be on "No, don't save".

Thoughts?

File: src/assets/scripts/Pinecone/Dialog/index.js [hunk 2]
```diff
@@ -61,6 +72,15 @@ class Dialog {
...
+			if ( this.config.inputDescription ) {
+				innerHtml += `<p class="input-group__description">${this.config.inputDescription}</p>`;
```
[the remark this answers is not in the document]
Comment (greatislander) [reply src/assets/scripts/Pinecone/Dialog/index.js]: Resolved in 91f28bf.

File: src/assets/scripts/Pinecone/Dialog/index.js (hunk 1 above)
Comment (greatislander) [reply src/assets/scripts/Pinecone/Dialog/index.js]: >     * Should there be an explicit close button on the dialog? My expectation is there should be one.

I thought we had discussed this at some point, but my thinking was that we shouldn't. I was thinking this for a few reasons:

1. The modal is essentially an enhanced version of a native [`confirm()`](https://codepen.io/greatislander/pen/RwNXvVX) dialog, which doesn't provide a close button; you either confirm or cancel (hitting the <kbd>esc</kbd> key is effectively the same as cancelling the operation).
2. This guide doesn't mention adding a close button other than the cancel button: https://guide.inclusivedesign.ca/tools/AccessibleDevelopmentTools.html
3. In this case, where a user may have entered text, closing the dialog with a neutral close button as opposed to a specific cancel button might lead some users to think that if they invoke the dialog again, their input will be preserved.

>     * If the input field is mandatory, I think the initial focus should be on the input field (or second in the tab order if a close button is added as per comment above).

Agreed! Fixed in c220f9b.
 
>     * If the input field is optional, the initial focus can be on "No, don't save".

**Pull request merged**